acmw-db-app/pages/forgotPassword.js

```diff
@@ -0,0 +1,99 @@
+import Head from 'next/head'
+import styles from '../styles/Reset.module.css'
+import React, { useState } from 'react';
+  
+class forgotPassword extends React.Component {
+    constructor() {
+    super();
+    this.state = {
+      input: {},
+      errors: {}
+    };
+     
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+     
+  handleChange(event) {
+    let input = this.state.input;
+    input[event.target.name] = event.target.value;
+  
+    this.setState({
+      input
+    });
+  }
+     
+  handleSubmit(event) {
+    event.preventDefault();
+  
+    if(this.validate()){
+        console.log(this.state);
+
+        let input = {};
+        input["email"] = "";
+        this.setState({input:input});
+  
+        alert('Email confirmed. Form "submitted"');
+    }
+  }
+  
+  validate(){
+      let input = this.state.input;
+      let errors = {};
+      let isValid = true;
+
+      if (!input["email"]) {
+        isValid = false;
+        errors["email"] = "Please enter your email Address.";
+      }
+
+      if (typeof input["email"] !== "undefined") {
+
+        var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
+        if (!pattern.test(input["email"])) {
+          isValid = false;
+          errors["email"] = "Please enter valid email address.";
```

Comment:
Lines 47 and 55 are basically the same and happen for essentially the same reason - the user is not giving valid input. I think removing those lines then adding `if(!isValid) { errors["email"] = "Please enter a valid email address" }` before line 59 would be cleaner

Reply:
I kept this and deleted the if statement above asking for a valid email.